### DISCUSSION TITLE
Electron to .NET session injection

### DISCUSSION THREAD

**Discussion by eishexac** [WalletConnect/walletconnect-monorepo#4020] [Developer Support] [answered]
I am currently developing a virtual reality project built on Ethereum, using the Unity engine. 

The project utilizes a launcher created with Electron, which establishes a successful connection to a web3 wallet via wagmi and the WalletConnect connector. 
After successful connection, I have observed that session-related data is being stored in the local storage within the Electron app. 
Additionally, my experiments with the .NET SDK using file storage have shown that the same identical session data is stored in a file.

The objective is to seamlessly integrate the WalletConnect session into the VR experience, allowing the user to authenticate only once. I am able to transfer the session data from the launcher to the VR environment through the use of sockets.

My inquiry pertains to the essential data required to inject the session into the .NET SDK effectively. 
I am curious if such an integration is feasible, and if so, how it could be implemented?

**Comment by skibitsky** [comment]
Hey @munchkinpayee,

I think using sockets locally is the way to go. You can create a custom storage implementation that communicates with the Electron app over a socket, and then pass this storage into `SignClient`.

- [FileSystemStorage](https://github.com/WalletConnect/WalletConnectSharp/blob/9dcc02c34163f4c274023bde6cb3ebea3fe0be2f/Core%20Modules/WalletConnectSharp.Storage/FileSystemStorage.cs) for reference
- [Example of passing custom storage into SignClient](https://github.com/WalletConnect/WalletConnectSharp/blob/2b5a8e32e3ba0d2d78543cac1a7532a42c707a3c/Tests/WalletConnectSharp.Sign.Test/SignClientFixture.cs#L21-L34)

**Reply by eishexac** [reply]
I intend to avoid using WalletConnect.Sign and instead utilize WalletConnect.Nethereum for blockchain interactions.
I see that WalletConnect.Nethereum is dependent on WalletConnect.Core.

I am unable to find documentation for the WalletConnect.Nethereum package. Is there any? 

I've successfully established a pairing between a wallet and the VR application using only WalletConnect.Core, as indicated by the [PairingAPI](https://docs.walletconnect.com/api/core/pairing) documentation.

From my understanding I already have a paired session inside the electron application, am I right?
If that is the case, how could I trick WalletConnect.Core into thinking it already has a paired session and where the session data is? 

I imagine I should do this using the received session data inside the .NET application.

`await sdkClient.core.pairing.activate({ topic: "1b3eda3f4..." })`
`await sdkClient.core.pairing.updateExpiry({ topic: "1b3eda3f4...", expiry: 60 })`
`await sdkClient.core.pairing.updateMetadata({ topic: "1b3eda3f4...", metadata: { name: "MyDapp", ... } })`

**Reply by skibitsky** [reply, marked as answer]
The `WalletConnect.Nethereum` package hasn't been updated for a while. As far as I know, it uses WalletConnect v1 which is deprecated. Upgrading it and providing better documentation for interacting with smart contracts using our packages is on our roadmap.

In the meantime, I'd suggest you use `WalletConnect.Sign` and `Nethereum` (directly).

> From my understanding I already have a paired session inside the electron application, am I right?

Correct.

> If that is the case, how could I trick WalletConnect.Core into thinking it already has a paired session and where the session data is?

After you have session in your storage you can do something like this to load it from there:
https://github.com/WalletConnect/WalletConnectUnity/blob/62e1c8bb35127957e02ac990df7abeaecca45fed/Packages/com.walletconnect.core/Runtime/WalletConnect.cs#L105

**Reply by eishexac** [reply]
Thank you for providing the references! What must be done is now perfectly understood. 
I will do as you suggested and use the SignClient in combination with Nethereum.

I greatly appreciate your support! Wishing you all the best!
